libbeat/processors/communityid/communityid.go

```diff
@@ -154,20 +154,22 @@ func (p *processor) buildFlow(event *beat.Event) *flowhash.Flow {
 		if err != nil {
 			return nil
 		}
-		flow.SourcePort, ok = tryToUint16(v)
-		if !ok || flow.SourcePort == 0 {
+		sp, ok := tryToUint(v)
+		if !ok || sp < 1 || sp > 65535 {
```

Comment:
I wonder if the fix would be enough with this extra checks without changing the `tryToUint16` function, even though I do not have a strong opinion against doing that change, either. WDYT @andrewkroh ?

Reply:
If you dont' modify the `tryToUint16`, any number that is larger than 65535 than the number is forced to be a unint16 which would then always be under 65535 so the conditionals don't work.

Reply:
> If you dont' modify the `tryToUint16`, any number that is larger than 65535 than the number is forced to be a unint16 which would then always be under 65535 so the conditionals don't work.

You are very correct, disregard my previous comment.